### PR TITLE
Add workflow for multi-arch-build, and prerequisites.

### PR DESCRIPTION
**Pull request by earthlingdavey**
This PR includes the work done here https://github.com/FreedomBen/dory-dnsmasq/pull/2 - it includes Mac M-series support and a makerfile (namely for local development).

It also uses the lates alpine release, and installs dnsmasq-dnssec within the Dockerfile, to reduce the external and outdated dependency on andyshinn/dnsmasq.

A workflow will build for multi-arch platforms and push to Docker Hub.

### CODE REVIEW

**Comment by earthlingdavey** [comment]
@wilson1000-MoJ , would you mind reviewing?
FYI. It looks like it worked nicely. Tested by deploying to my personal DockerHub image

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,10 @@
-FROM andyshinn/dnsmasq:2.75
+ARG alpine_tag
+
+FROM alpine:${alpine_tag}
+
+ARG dnsmasq_version
+RUN apk --no-cache add dnsmasq-dnssec=${dnsmasq_version}
```
Comment (wilson1000-MoJ) [inline Dockerfile:6]: Does APK allow version specific calls on `add`?
```
dnsmasq-dnssec=${dnsmasq_version}
```

File: README.md [hunk 2]
```diff
@@ -62,3 +62,38 @@ docker run -p 53:53/tcp -p 53:53/udp --cap-add=NET_ADMIN freedomben/dory-dnsmasq
...
+Remove the builder:
+
+```bash
+make remove
+```
+
+### Releases
+
+A GitHub workflow is used to build and push a container image to Docker Hub.
+- Before release, ensure that you have ound a valid version of dnsmasq-dnssec, on the [Alpine Packages page](https://pkgs.alpinelinux.org/packages?name=dnsmasq-dnssec&branch=v3.19&repo=&arch=&maintainer=).
```
Comment (wilson1000-MoJ) [inline README.md]: Small typo :o) : `ound`

**Review by wilson1000-MoJ — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (earthlingdavey) [reply Dockerfile:6]: Yes, I thought, even though we didn't talk it through... it would be nice if the docker tag tracked the version of dnsmasq.

**Review by wilson1000-MoJ — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (earthlingdavey) [reply README.md]: Fixed, thanks.

**Comment by earthlingdavey** [comment]
Fingers crossed for the workflow. Obviously, it will never work first time!

**Pull request merged**